js/web/lib/wasm/jsep/webgpu/ops/attention.ts

```diff
@@ -652,10 +652,25 @@ export const applyAttention = (
   parameters: AttentionParameters,
   attributes: AttentionAttrs,
 ) => {
-  const outputCount = context.outputCount;
-  const pastSequenceLength = parameters.kvNumHeads !== undefined || outputCount > 1 ? parameters.pastSequenceLength : 0;
+  const pastSequenceLength =
+    parameters.kvNumHeads !== undefined || context.outputCount > 1 ? parameters.pastSequenceLength : 0;
   const totalSequenceLength = pastSequenceLength + parameters.kvSequenceLength;
```

Comment:
It is able to get total sequence length from parameters. No need to recompute here.

Reply:
Also, I do not understand that why we need update pastSequenceLength. For example, even when outputCount is 1 (do not output present), the length in the past (or past_key) input shall not be reset to zero since past exists.

Reply:
I actually don't care about any of the seqlen, but need it to pass the correct shape to context.output()

Reply:
`parameters.totalSequenceLength` can be used for shape.

BTW, pastSequenceLength is passed to createAttentionProbsProgramInfo:
https://github.com/microsoft/onnxruntime/blob/63e88499923cc89bb01b9b6844ffe7bb79a0149d/js/web/lib/wasm/jsep/webgpu/ops/attention.ts#L354-L356
That's not needed as well since we can use parameters.totalSequenceLength.

Reply:
ok, using parameters.totalSequenceLength for this PR.
But we need to go and review this entire file and mha later.